File: Leetcode 2038. Remove Colored Pieces if Both Neighbors are the Same Color/lc 2038.java
class Solution {
    public boolean winnerOfGame(String colors) {
        int aC=0;
        int bC=0;
        
        int i=0;
        while(i<colors.length()){
            int t1=0;
            int t2=0;
            while(i<colors.length() && colors.charAt(i)=='A'){
                t1++;
                if(t1>=3){
                    aC++;
                }
                i++;
            }
            while(i<colors.length() && colors.charAt(i)=='B'){
                t2++;
                if(t2>=3){
                    bC++;
                }
                i++;
            }
        }

        if(aC>bC){
            return true;
        }else{
            return false;
        }
    }
}

File: Leetcode 2038. Remove Colored Pieces if Both Neighbors are the Same Color/lc 2038.cpp
class Solution {
public:
    bool winnerOfGame(string colors) {
        int aC=0,bC=0;
        int i=0;

        while(i<colors.size()){
            int t1=0,t2=0;

            while(i<colors.size() && colors[i]=='A'){
                t1++;
                if(t1>=3){
                    aC++;
                }
                i++;
            }

            while(i<colors.size() && colors[i]=='B'){
                t2++;
                if(t2>=3){
                    bC++;
                }
                i++;
            }

        }

        return (aC>bC)?true:false;
    }
};